iothub_client/samples/pnp/pnp_simple_thermostat/pnp_simple_thermostat.c

```diff
@@ -27,14 +27,50 @@
 #include "certs.h"
 #endif // SET_TRUSTED_CERT_IN_SAMPLES
 
+#ifdef USE_PROV_MODULE_FULL
+#include "pnp_dps_ll.h"
+#endif // USE_PROV_MODULE_FULL
+
 // JSON parser library
 #include "parson.h"
 
+// Environment variable used to specify how app connects to hub and the two possible values
+static const char g_securityTypeEnvironmentVariable[] = "IOTHUB_DEVICE_SECURITY_TYPE";
+static const char g_securityTypeConnectionStringValue[] = "connectionString";
+static const char g_securityTypeDpsValue[] = "DPS";
+
 // Environment variable used to specify this application's connection string
 static const char g_connectionStringEnvironmentVariable[] = "IOTHUB_DEVICE_CONNECTION_STRING";
 
-// Amount of time to sleep between sending telemetry to IotHub, in milliseconds.  Set to 1 minute.
-static unsigned int g_sleepBetweenTelemetrySends = 60 * 1000;
+#ifdef USE_PROV_MODULE_FULL
+// Environment variable used to specify this application's DPS id scope
+static const char g_dpsIdScopeEnvironmentVariable[] = "IOTHUB_DEVICE_DPS_ID_SCOPE";
+
+// Environment variable used to specify this application's DPS device id
+static const char g_dpsDeviceIdEnvironmentVariable[] = "IOTHUB_DEVICE_DPS_DEVICE_ID";
+
+// Environment variable used to specify this application's DPS device key
+static const char g_dpsDeviceKeyEnvironmentVariable[] = "IOTHUB_DEVICE_DPS_DEVICE_KEY";
+
+// Environment variable used to optionally specify this application's DPS id scope
+static const char g_dpsEndpointEnvironmentVariable[] = "IOTHUB_DEVICE_DPS_ENDPOINT";
+
+// Global provisioning endpoint for DPS if one is not specified via the environment
+static char g_dps_DefaultGlobalProvUri[] = "global.azure-devices-provisioning.net";
+
+// Values of connection / security settings read from environment variables and/or DPS runtime
+PNP_DEVICE_CONFIGURATION g_pnpDeviceConfiguration;
+#endif // USE_PROV_MODULE_FULL
+
+// Connection string used to authenticate device when connection strings are used
+const char* g_pnpDeviceConnectionString;
+
+// Amount of time to sleep between polling hub, in milliseconds.  Set to wake up every 100 milliseconds.
+static unsigned int g_sleepBetweenPolls = 100;
+
+// Every time the main loop wakes up, on the g_sendTelemetryFrequency(th) pass will send a telemetry message.
+// So we will send telemetry every (g_sendTelemetryFrequency * g_sleepBetweenPolls) milliseconds; 60 seconds as currently configured.
+static const int g_sendTelemetryFrequency = 600;
```

Comment:
```suggestion
static const int g_sendTelemetryFrequencyMs = 600;
```

Reply:
This actually isn't in milliseconds.  It's basically every time through the main() loop, we'll DoWork() each time and every 600 do a send telemetry.  This lets us do a simple modulo on a counter, instead of having to do any math on time.

I agree "frequency" can mean too many things to different people.  I'm going with `g_sendTelemetryPollInterval` to try and capture that we're tied to polling and so people don't read into this as much.